logging.js

```diff
@@ -50,11 +56,28 @@ export class ServerLogger {
 		if (!this._loggerPromise && !navigator || !navigator.sendBeacon) {
 			return;
 		}
-		const logger = await this._loggerPromise;
 		if (this._logs.length > 0) {
-			const data = JSON.stringify(this._logs);
-			navigator.sendBeacon(logger.Endpoint, data);
+			try {
+				const logger = await this._loggerPromise;
+				const data = JSON.stringify(this._logs);
+				navigator.sendBeacon(logger.Endpoint, data);
+			} catch (err) {
+				console.error(err, this._logs);
+			}
+		}
+	}
+
+	async _provisionLoggerEndpoint(reload) {
+		const htmlEle = document.getElementsByTagName('html')[0];
+		if (!htmlEle) {
+			throw new Error(`Failed to locate top-level HTML element for ${dataLoggingEndpointAttribute}`);
```

Comment:
Could these result in infinite errors if there's a `window.onerror` that calls into this code?

Reply:
`_provisionLoggerEndpoint` is only ever `await`'ed in a `try` block so I think we should be safe there.